crates/bevy_pbr/src/render/mesh/mod.rs

```diff
@@ -1126,33 +1158,45 @@ impl<P: PhaseItem, const I: usize> RenderCommand<P> for SetMeshViewBindGroup<I>
 
 pub struct SetMeshBindGroup<const I: usize>;
 impl<P: PhaseItem, const I: usize> RenderCommand<P> for SetMeshBindGroup<I> {
-    type Param = SRes<MeshBindGroup>;
+    type Param = SRes<RenderAssets<Mesh>>;
     type ViewWorldQuery = ();
     type ItemWorldQuery = (
+        Read<Handle<Mesh>>,
         Read<DynamicUniformIndex<MeshUniform>>,
         Option<Read<SkinnedMeshJoints>>,
+        Option<Read<morph::Index>>,
     );
+
     #[inline]
     fn render<'w>(
         _item: &P,
         _view: (),
-        (mesh_index, skinned_mesh_joints): ROQueryItem<'_, Self::ItemWorldQuery>,
-        mesh_bind_group: SystemParamItem<'w, '_, Self::Param>,
+        (mesh, mesh_index, skin_index, morph_index): ROQueryItem<Self::ItemWorldQuery>,
+        meshes: SystemParamItem<'w, '_, Self::Param>,
         pass: &mut TrackedRenderPass<'w>,
     ) -> RenderCommandResult {
-        if let Some(joints) = skinned_mesh_joints {
-            pass.set_bind_group(
-                I,
-                mesh_bind_group.into_inner().skinned.as_ref().unwrap(),
-                &[mesh_index.index(), joints.index],
+        let Some(gpu_mesh) = meshes.into_inner().get(mesh) else {
+            error!(
+                "An entity with the DynamicUniformIndex<mesh::Uniform> had a Handle<Mesh> not \
+                present in its RenderAssets<Mesh> (very unexpected)"
             );
-        } else {
-            pass.set_bind_group(
-                I,
-                &mesh_bind_group.into_inner().normal,
-                &[mesh_index.index()],
+            return RenderCommandResult::Failure;
+        };
+        let Some(bind_group) = &gpu_mesh.bind_group else {
+            error!(
+                "The bind_group field of GpuMesh wasn't set in the render phase. This is odd, \
+                since it should be set by the queue_mesh_bind_group system."
```

Comment:
Does this case have any chance to happen? If not, a comment and ignoring it would be better. 

If it does, the message shouldn't say "this is odd", but something like "this is a bug in Bevy, please open an issue"

Reply:
Yeah good idea, it's so much better to have actionable messages :sweat_smile: I'm a bit annoyed I didn't think of it.

I don't think it's possible to hit this line in _current_ bevy. But since this condition depends on a lot of complexly linked code, it's believable that it can be hit in the future.